CRM/Core/BAO/CustomField.php

```diff
@@ -203,7 +203,8 @@ public static function create(&$params) {
         'table_name'
       );
 
-      if ($params['option_type'] == 1 && empty($params['option_group_id'])) {
+      //CRM-16659: if option_value then create an option group for this custom field.
+      if ($params['option_type'] == 1 && (empty($params['option_group_id']) || !empty($params['option_value']))) {
```

Comment:
Please check that there is another check for $params['option_value'] inside the if condition again.